internal/gengapic/genrest.go

```diff
@@ -0,0 +1,179 @@
+	p("// It satisfies the internal%sClient interface.", servName)
+	p("//")
+	p("// Methods, except Close, may be called concurrently. However, fields must not be modified concurrently with method calls.")
+	p("type %sRestClient struct {", lowcaseServName)
+	p("  host string")
+	p("}")
+	p("")
+	g.restClientUtilities(serv, servName, imp, hasRPCForLRO)
+	g.genRESTMethods(serv, servName)
+}
+
+func (g *generator) genRESTMethods(serv *descriptor.ServiceDescriptorProto, servName string) error {
+	// Clear LROs between services
+	// TODO: handle LRO types shared between rest and grpc clients with a map[bool]
+	g.addMetadataServiceForTransport(serv.GetName(), "rest", servName)
+
+	methods := append(serv.GetMethod(), g.getMixinMethods()...)
+
+	for _, m := range methods {
+		g.methodDoc(m)
+		if err := g.genRESTMethod(servName, serv, m); err != nil {
+			return errors.E(err, "method: %s", m.GetName())
+		}
+		g.addMetadataMethod(serv.GetName(), "rest", m.GetName())
+	}
+
+	sort.Slice(g.aux.lros, func(i, j int) bool {
+		return g.aux.lros[i].GetName() < g.aux.lros[j].GetName()
+	})
+	for _, m := range g.aux.lros {
+		if err := g.lroType(servName, serv, m); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (g *generator) restClientUtilities(serv *descriptor.ServiceDescriptorProto, servName string, imp pbinfo.ImportSpec, hasRPCForLRO bool) {
+	// p := g.printf
+
+	// clientName := camelToSnake(serv.GetName())
+	// clientName = strings.Replace(clientName, "_", " ", -1)
+
+	// NOTE(dovs): enable this function only when we can verify that it works.
+	// If we leave it commented out, we can change it, and we don't have to
+	// guarantee correctness or stability.
+	//
+	// p("// New%sRestClient creates a new %s rest client.", servName, clientName)
+	// p("//")
+	// g.comment(g.comments[serv])
+	// p("func New%[1]sRestClient(ctx context.Context, opts ...option.ClientOption) (*%[1]sRestClient, error) {", servName)
+	// p("    c := &%sRestClient{", servName)
+	// p("    }")
+	// p("")
+	// p("    return &%sClient{internal%[1]sClient: c, CallOptions: default%[1]sCallOptions()}, nil", servName)
+	// p("}")
+
+}
+
+// genRESTMethod generates a single method from a client. m must be a method declared in serv.
+// If the generated method requires an auxiliary type, it is added to aux.
+func (g *generator) genRESTMethod(servName string, serv *descriptor.ServiceDescriptorProto, m *descriptor.MethodDescriptorProto) error {
+	if g.isLRO(m) {
+		// TODO(dovs)
+		return nil
+	}
+
+	if m.GetOutputType() == emptyType {
+		// TODO(dovs)
+		return nil
+	}
+
+	if pf, err := g.pagingField(m); err != nil {
+		return err
+	} else if pf != nil {
+		// TODO(dovs)
+		return nil
+	}
+
+	switch {
+	case m.GetClientStreaming():
+		// TODO(dovs)
+		return nil
+	case m.GetServerStreaming():
+		// TODO(dovs)
+		return nil
+	default:
+		// TODO(dovs)
+		return g.unaryRestCall(servName, m)
+	}
+}
+
+func doAThing(m *descriptor.MethodDescriptorProto) (error, error) {
```

Comment:
Update function? Though, it appears this function doesn't have any effects?

Reply:
This is incremental development and debug cruft that should be removed.